src/racer/core.rs

```diff
@@ -97,6 +97,15 @@ pub struct Match {
     pub docs: String,
 }
 
+impl Match {
+    /// Checks if two matches can be considered the same for deduplication purposes.
+    fn is_same_as(&self, other: &Match) -> bool {
```

Comment:
Is there a reason we don't just use PartialEq/Eq here? From the perspective of the user, they are identical.